### PR TITLE
Do not create directories . and ..

### LINKED ISSUE #1647: solc crashes on "-o ."
```
$ solc/solc --asm -o . xz.sol 
Warning: This is a pre-release compiler version, please do not use it in production.
xz.sol:1:1: Warning: Source file does not specify required compiler version!
contract A {
^
Spanning multiple lines.
libc++abi.dylib: terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::create_directories: Invalid argument
Abort trap: 6
```

Calling  `create_directories(".")` throws an exception, same applies to `..` and any combinations of them. It does work however if the any path item is not one of them, i.e. `solc/.` or `solc/..` or `./solc`.

### PR DESCRIPTION
**Pull request by axic**
Fixes #1647.

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -461,7 +461,10 @@ void CommandLineInterface::createFile(string const& _fileName, string const& _da
 	namespace fs = boost::filesystem;
 	// create directory if not existent
 	fs::path p(m_args.at(g_argOutputDir).as<string>());
-	fs::create_directories(p);
+	// Do not try creating the directory if the first item is . or ..
+	if (p.filename() != "." && p.filename() != "..") {
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Style: either put `{` on its own line or remove the `{` `}` pair altogether.

Comment (axic) [reply solc/CommandLineInterface.cpp]: Fixed.

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Conflicts in changelog

**Comment by chriseth** [comment]
Conflict

**Pull request merged**